frontend/src/toolbar/utils.ts

```diff
@@ -41,14 +41,19 @@ export function elementToQuery(element: HTMLElement, dataAttributes: string[]):
             continue
         }
 
-        const selector = `[${cssEscape(name)}="${cssEscape(value)}"]`
-        if (querySelectorAllDeep(selector).length == 1) {
-            return selector
+        const escapedSelector = `[${cssEscape(name)}="${cssEscape(value)}"]`
+        const unescapedSelector = `[${name}="${value}"]`
+
+        if (querySelectorAllDeep(escapedSelector).length == 1) {
```

Comment:
How the validness of CSS is determined by the `querySelectorAllDeep` result? Is it just a regular `querySelectorAll` version, but the one that goes through shadow DOM also?

Does it mean that the validity of the selector is determined if only one element is matched? But why return an unescaped selector then instead of the escaped one we just tested?

I see the comment below, but I lack the context to understand it.

Reply:
"how is the validness of CSS determined by querySelectorAllDeep" - I'd never dug in... turns out it's a library https://github.com/Georgegriff/query-selector-shadow-dom#readme

My assumption is that it's using browser apis to match the query so it's valid if the browser says it is valid :)

> why return an unescaped selector then instead of the escaped one we just tested?

the escaped selector is valid but the bug we discovered is that our action matching can't use all valid css, so we're gambling that it's better to use the unescaped selector than not (since we now know there are some cases where escaped simply won't work in actions)